crypto/cmp/cmp_asn.c

```diff
@@ -287,23 +287,31 @@ OSSL_CMP_ITAV *OSSL_CMP_ITAV_new_rootCaKeyUpdate(const X509 *newWithNew,
                                                  const X509 *oldWithNew)
 {
     OSSL_CMP_ITAV *itav;
-    OSSL_CMP_ROOTCAKEYUPDATE *upd = OSSL_CMP_ROOTCAKEYUPDATE_new();
+    OSSL_CMP_ROOTCAKEYUPDATE *upd = NULL;
 
+    if (newWithNew == NULL)
+        goto null_value;
+
+    upd = OSSL_CMP_ROOTCAKEYUPDATE_new();
     if (upd == NULL)
         return NULL;
-    if (newWithNew != NULL && (upd->newWithNew = X509_dup(newWithNew)) == NULL)
+
+    if ((upd->newWithNew = X509_dup(newWithNew)) == NULL)
         goto err;
     if (newWithOld != NULL && (upd->newWithOld = X509_dup(newWithOld)) == NULL)
         goto err;
     if (oldWithNew != NULL && (upd->oldWithNew = X509_dup(oldWithNew)) == NULL)
         goto err;
+
+ null_value:
```

Comment:
Better not use `goto` for anything else but error handling.
So please use `if (newWithNew != NULL) { ... }`.

Reply:
done